lms/views/api/canvas/authorize.py

```diff
@@ -0,0 +1,48 @@
+import secrets
+from urllib.parse import urlencode, urlparse, urlunparse
+
+from pyramid.httpexceptions import HTTPFound
+from pyramid.view import view_config
+
+from lms.validation import CANVAS_OAUTH_CALLBACK_SCHEMA
+
+
+@view_config(route_name="canvas_api_authorize", request_method="GET")
+def authorize(request):
+    ai_getter = request.find_service(name="ai_getter")
+    consumer_key = request.lti_user.oauth_consumer_key
+
+    authorize_url = urlunparse(
+        (
+            "https",
+            urlparse(ai_getter.lms_url(consumer_key)).netloc,
```

Comment:
Will it ever be the case that the Canvas root URL has a path component, rather than only a host?

Reply:
Yep, that will sometimes be the case. In fact, you can literally type anything into the `/welcome` form and it'll get saved in the DB! So a future PR is gonna have to add robust validation of these as we read them _out_ of the DB.